libs/vkd3d/command.c

```diff
@@ -14669,11 +14692,337 @@ static void STDMETHODCALLTYPE d3d12_command_list_SetProtectedResourceSession(d3d
     FIXME("iface %p, protected_session %p stub!\n", iface, protected_session);
 }
 
+static void vk_clear_color_value_from_d3d12(VkClearColorValue *vk_value, const D3D12_CLEAR_VALUE *d3d_value)
+{
+    unsigned int i;
+
+    for (i = 0; i < 4; i++)
+        vk_value->float32[i] = d3d_value->Color[i];
+}
+
+static void vk_clear_depth_stencil_value_from_d3d12(VkClearDepthStencilValue *vk_value,
+        const D3D12_CLEAR_VALUE *d3d_depth_value, const D3D12_CLEAR_VALUE *d3d_stencil_value)
+{
+    vk_value->depth = d3d_depth_value->DepthStencil.Depth;
+    vk_value->stencil = d3d_stencil_value->DepthStencil.Stencil;
+}
+
+static VkAttachmentLoadOp vk_load_op_from_d3d12(D3D12_RENDER_PASS_BEGINNING_ACCESS_TYPE access_type)
+{
+    if (access_type == D3D12_RENDER_PASS_BEGINNING_ACCESS_TYPE_CLEAR)
+        return VK_ATTACHMENT_LOAD_OP_CLEAR;
+
+    if (access_type == D3D12_RENDER_PASS_BEGINNING_ACCESS_TYPE_DISCARD)
+        return VK_ATTACHMENT_LOAD_OP_DONT_CARE;
+
+    return VK_ATTACHMENT_LOAD_OP_LOAD;
+}
+
+static void d3d12_command_list_load_render_pass_rtv(struct d3d12_command_list *list,
+        struct d3d12_rtv_desc *rtv_info, const D3D12_RENDER_PASS_RENDER_TARGET_DESC *rt)
+{
+    VkAttachmentLoadOp load_op = vk_load_op_from_d3d12(rt->BeginningAccess.Type);
+    VkClearValue clear_value;
+
+    vk_clear_color_value_from_d3d12(&clear_value.color, &rt->BeginningAccess.Clear.ClearValue);
+
+    d3d12_command_list_track_resource_usage(list, rtv_info->resource,
```

Comment:
The RTV must cover the full resource to be able to signal false here. There should be a helper vkd3d_rtv_and_aspects_fully_cover_resource that's used by d3d12_command_list_clear_attachment.

Reply:
urg, I hate this. Anyway, should be fixed.